FindmallocMC.cmake

```diff
@@ -41,66 +45,48 @@
 ###############################################################################
 
 
-###############################################################################
-# Required cmake version
-###############################################################################
-
+# Required cmake version ######################################################
+#
 cmake_minimum_required(VERSION 2.8.5)
 
 
-###############################################################################
-# mallocMC
-###############################################################################
-
-# we start by assuming we found mallocMC and falsify it if some
-# dependencies are missing (or if we did not find mallocMC at all)
+# mallocMC ####################################################################
+#
 set(mallocMC_FOUND TRUE)
 
 
-###############################################################################
-# preconditions
-###############################################################################
-
-if(NOT CUDA_FOUND)
-    set(mallocMC_FOUND FALSE)
-    message(STATUS "could not find CUDA, try something like find_package(CUDA REQUIRED)")
-elseif(CUDA_VERSION VERSION_LESS "5.0")
-    set(mallocMC_FOUND FALSE)
-    message(STATUS "CUDA found, but version too low (needs 5.0 or higher)")
-endif(NOT CUDA_FOUND)
-
-if(NOT Boost_FOUND)
-    set(mallocMC_FOUND FALSE)
-    message(STATUS "could not find Boost, try something like find_package(Boost REQUIRED)")
-elseif(Boost_VERSION LESS 104800)
-    set(mallocMC_FOUND FALSE)
-    message(STATUS "Boost found, but version too low (needs 1.48 or higher)")
-endif(NOT Boost_FOUND)
+# dependencies ################################################################
+#
+find_package(CUDA 5.0 REQUIRED)
+find_package(Boost 1.48.0 REQUIRED)
 
 
-# find mallocMC installation #################################################
+# find mallocMC installation ##################################################
 #
 find_path(mallocMC_ROOT_DIR
-  NAMES include/mallocMC/mallocMC.hpp
-  PATHS ENV MALLOCMC_ROOT
-  DOC "mallocMC ROOT location"
-)
+    NAMES include/mallocMC/mallocMC.hpp
+    PATHS ENV MALLOCMC_ROOT
```

Comment:
Could you add a possibility to provide the MallocMC root directory through a CMake variable in addition to the environment variable?
This would simplify the usage example given in your description and would make it more consistent to other Find scripts e.g. FindBoost with its BOOST_ROOT.
Furthermore this would allow to compile a project using MallocMC with different MallocMC variants only by changing the cmake command line.